### PR TITLE
Convert perf.groovy to use run-xunit-perf.py

### LINKED ISSUE #12143: CoreFx Linux Performance baseline should be updated automatically
Currently, the commit hash is hardcoded in this line:
https://github.com/dotnet/coreclr/blob/master/tests/scripts/perf-prep.sh#L76

### PR DESCRIPTION
**Pull request by adiaaida**
Also removes run-xunit-perf.cmd and run-xunit-perf.sh. This change moves us to using build-tests.sh to generate the layout (ie pull down corefx), so this change fixes #12143.

### CODE REVIEW

**Comment by adiaaida** [comment]
@dotnet-bot test Windows_NT x64 perf please
@dotnet-bot test linux perf flow please
@dotnet-bot test Windows_NT x64 illink
@dotnet-bot test Windows_NT x64 perf scenarios

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet/rap-team PTAL

(ignore the perf test failures -- I forgot you had to enable the generated jobs and run those)

**Comment by adiaaida** [comment]
@jorive that is here: https://ci2.dot.net/job/dotnet_coreclr/job/perf/job/master/job/GenPRTest/job/generator_prtest/764/

**Comment by adiaaida** [comment]
but I can run it again.
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by kasper3** [comment]
Assuming this step kicks in after the managed code compilation and we have roslyn compilers always available on host platform at that point, wouldn't it be a good idea to use C# scripts in all post-build scenarios like in #15858, instead of relying on python, batch or bash?

**Comment by adiaaida** [comment]
@kasper3 For now, we do not have plans on porting this testing infrastructure to C# scripts. This change is simply updating our groovy scripting from the cmd/sh scripts to a unified python script we have already written.

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

**Comment by jorive** [comment]
Will this PR solve issue: https://github.com/dotnet/coreclr/issues/12143?

**Comment by adiaaida** [comment]
@dotnet-bot test ci please

File: tests/scripts/run-xunit-perf.py [hunk 1]
```diff
@@ -149,7 +149,7 @@ def run_command(runArgs, environment, errorMessage):
         subprocess.check_output(runArgs, stderr=subprocess.PIPE, env=environment)
     except subprocess.CalledProcessError as e:
         log(e.output.decode('utf-8'))
-        raise RuntimeException(errorMessage);
+        raise RuntimeError(errorMessage);
```
Comment (jorive) [inline tests/scripts/run-xunit-perf.py]: Could you just raise the caught exception?

Comment (jorive) [reply tests/scripts/run-xunit-perf.py]: You do not need the `;` at the end of the line

Comment (adiaaida) [reply tests/scripts/run-xunit-perf.py]: Sure.

**Comment by adiaaida** [comment]
All of the legs are passing with the new scripting now.

File: tests/scripts/run-xunit-perf.py [hunk 2]
```diff
@@ -149,7 +149,7 @@ def run_command(runArgs, environment, errorMessage):
         subprocess.check_output(runArgs, stderr=subprocess.PIPE, env=environment)
     except subprocess.CalledProcessError as e:
         log(e.output.decode('utf-8'))
```
Comment (jorive) [inline tests/scripts/run-xunit-perf.py]: Should we be doing this:
```python
except subprocess.CalledProcessError as e: 
        log(errorMessage)  # errorMessage(str): message to print if there is an error
        log(e.output.decode('utf-8')) 
        raise e
```

Comment (adiaaida) [reply tests/scripts/run-xunit-perf.py]: I can, if you'd prefer it that way.

Comment (jorive) [reply tests/scripts/run-xunit-perf.py]: Yes, please. Let's do this (Bubble up the caught exception).

Comment (adiaaida) [reply tests/scripts/run-xunit-perf.py]: done

**Pull request merged**